hyperspy/misc/holography/reconstruct.py

```diff
@@ -0,0 +1,207 @@
+
+        Returns
+        -------
+        Tuple of the sideband position (y, x), referred to the unshifted FFT.
+        """
+
+        sb_position = (0, 0)
+
+        f_freq = freq_array(holo_data.shape, holo_sampling)
+
+        # If aperture radius of centerband is not given, it will be set to 5 % of the Nyquist
+        # frequency.
+        if central_band_mask_radius is None:
+            central_band_mask_radius = 1 / 20. * np.max(f_freq)
+
+        # A small aperture masking out the centerband.
+        aperture_central_band = np.subtract(1, aperture_function(f_freq, central_band_mask_radius, 0))
+
+        fft_holo = fft2(holo_data) / np.prod(holo_data.shape)
+        fft_filtered = fft_holo * aperture_central_band
+
+        # Sideband position in pixels referred to unshifted FFT
+        if sb == 'lower':
+            fft_sb = fft_filtered[:int(fft_filtered.shape[0] / 2), :]
+            sb_position = np.asarray(np.unravel_index(fft_sb.argmax(), fft_sb.shape))
+        elif sb == 'upper':
+            fft_sb = fft_filtered[int(fft_filtered.shape[0] / 2):, :]
+            sb_position = (np.unravel_index(fft_sb.argmax(), fft_sb.shape))
+            sb_position = np.asarray(np.add(sb_position, (int(fft_filtered.shape[0] / 2), 0)))
+
+        return sb_position
+
+
+def find_sideband_size(holo_shape, sb_position, sb_size_ratio=0.5):
+    """
+    Finds the size of sideband filter
+
+    Parameters
+    ----------
+    holo_shape : array_like
+            Holographic data array
+    sb_position : tuple
+        The sideband position (y, x), referred to the non-shifted FFT.
+    sb_size_ratio : float, optional
+        Size of sideband as a fraction of the distance to central band
+
+    Returns
+    -------
+    sb_size : float
+        Size of sideband filter
+
+    """
+
+    h = np.array((np.asarray(sb_position) - np.asarray([0, 0]),
+                         np.asarray(sb_position) - np.asarray([0, holo_shape[1]]),
+                         np.asarray(sb_position) - np.asarray([holo_shape[0], 0]),
+                         np.asarray(sb_position) - np.asarray(holo_shape))) * sb_size_ratio
+    return np.min(np.linalg.norm(h, axis=1))
+
+
+def reconstruct(holo_data, holo_sampling, sb_size, sb_position, sb_smoothness, output_shape=None,
+                     plotting=False):
+        """Core function for holographic reconstruction.
+
+        Parameters
+        ----------
+        holo_data : array_like
+            Holographic data array
+        holo_sampling : tuple
+            Sampling rate of the hologram in y and x direction.
+        sb_size : float
+            Size of the sideband filter in pixel.
+        sb_position : tuple
+            Sideband position in pixel.
+        sb_smoothness: float
+            Smoothness of the aperture in pixel.
+        output_shape: tuple, optional
+            New output shape.
+        plotting : boolean
+            Shows details of the reconstruction (i.e. SB selection).
```

Comment:
At the moment it is plotting nothing, but we should think about what we want to plot.